/****************************************************************************
//	Usagi Engine, Copyright Â© Vitei, Inc. 2013
****************************************************************************/
import 'nanopb.proto';
import 'Engine/Maths/Maths.proto';

package usg.text;


message CharDefinition
{
	required uint32 CharData = 1;
	required Vector2f UV_TopLeft = 2;
	required Vector2f UV_BottomRight = 3;
};


message FontDefinition
{
	required string FontName = 1 [(nanopb).max_size = 32];
	required float DrawScale = 2;	// To account for padding and offsets
	required float LowerOffset = 3;
	required float Spacing = 4;
	repeated CharDefinition Chars = 7 [(nanopb).decoder = "PBChunkedArray<usg::ALLOC_OBJECT, 32>"];
};

